rpm/0007-make-unicode_printable.patch

```diff
@@ -0,0 +1,49 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Matti Viljanen <matti.viljanen@kapsi.fi>
+Date: Tue, 13 Sep 2022 18:21:04 +0300
+Subject: [PATCH] make high ASCII chars printable
+
+Currently busybox utils like "ls" fail to display filenames containing UTF-8
+characters, replacing any special characters with "?".
+
+Change libbb's printable_string() function to allow high ASCII characters so
+that unicode filenames are displayed correctls.
+
+Sailfish OS Forum thread:
+https://forum.sailfishos.org/t/asian-character-support-in-terminal/12938
+
+Cherry-picked from OpenWRT:
+https://git.openwrt.org/?p=openwrt/openwrt.git;a=commit;h=2ab0dae9915d5c7c1c06c26ce1956bf0badd7a5b
+
+Oritinal issue and changeset:
+https://dev.archive.openwrt.org/ticket/7993.html
+https://dev.archive.openwrt.org/changeset/43084.html
+---
+ libbb/printable_string.c | 4 +---
+ 1 file changed, 1 insertion(+), 3 deletions(-)
+
+diff --git a/libbb/printable_string.c b/libbb/printable_string.c
+index a814fd03c..f4c7b000d 100644
+--- a/libbb/printable_string.c
++++ b/libbb/printable_string.c
+@@ -28,8 +28,6 @@ const char* FAST_FUNC printable_string2(uni_stat_t *stats, const char *str)
+ 		}
+ 		if (c < ' ')
+ 			break;
+-		if (c >= 0x7f)
+-			break;
+ 		s++;
+ 	}
+ 
+@@ -42,7 +40,7 @@ const char* FAST_FUNC printable_string2(uni_stat_t *stats, const char *str)
+ 			unsigned char c = *d;
+ 			if (c == '\0')
+ 				break;
+-			if (c < ' ' || c >= 0x7f)
++			if (c < ' ')
+ 				*d = '?';
+ 			d++;
+ 		}
+-- 
+2.37.3
```

Comment:
please generate patch with --no-signature to not have the trailing git version in the path.

Reply:
Thanks for the tip, I'll keep that in mind for any future patches!